addons/medical_engine/functions/fnc_updateDamageEffects.sqf

```diff
@@ -33,12 +32,12 @@ if (EGVAR(medical,fractures) > 0) then {
     if ((_fractures select 2) == 1) then { _aimFracture = _aimFracture + 4; };
     if ((_fractures select 3) == 1) then { _aimFracture = _aimFracture + 4; };
 
-    if (EGVAR(medical,fractures) == 2) then { // the limp with a splint will still cause effects
-        private _isSprintBlocked = ((_fractures select 4) == -1) || {(_fractures select 5) == -1}; // block sprinting if we have a leg splint on
-        if (_isSprintBlocked || {!isSprintAllowed _unit}) then { // only update status effect if we need to
-            TRACE_1("updating status effect",_isSprintBlocked);
-            [_unit, "blockSprint", QEGVAR(medical,fracture), _isSprintBlocked] call EFUNC(common,statusEffect_set);
-        };
+    // Block sprint / force walking based on fracture setting and leg splint status
+    private _hasLegSplint = (_fractures select 4) == -1 || {(_fractures select 5) == -1};
+    [_unit, "blockSprint", QEGVAR(medical,fracture), _hasLegSplint && {EGVAR(medical,fractures) == 2}] call EFUNC(common,statusEffect_set);
+    [_unit, "forceWalk", QEGVAR(medical,fracture), _hasLegSplint && {EGVAR(medical,fractures) == 3}] call EFUNC(common,statusEffect_set);
+
+    if (EGVAR(medical,fractures) in [2, 3]) then { // the limp with a splint will still cause effects
```

Comment:
I think this could be back on line 36 still, no reason to run this code on settings 0/1
also "only update status effect if we need to" is still a good idea as statusEffect_set has some overhead